packages/api/test/indexer.test.ts

```diff
@@ -0,0 +1,154 @@
+import { TRPCError } from "@trpc/server";
+import type { inferProcedureInput } from "@trpc/server";
+import exp from "constants";
+import { describe, expect, it, vi } from "vitest";
+
+import { prisma } from "@blobscan/db";
+
+import type { AppRouter } from "../src/root";
+import { getCaller, INDEXER_DATA } from "./helpers";
+
+vi.mock("../src/env", () => ({
+  env: {
+    SECRET_KEY: "supersecret",
+  },
+}));
+
+vi.mock("@blobscan/blob-storage-manager/src/env", () => ({
+  env: {
+    CHAIN_ID: 1,
+    POSTGRES_STORAGE_ENABLED: true,
+    GOOGLE_STORAGE_ENABLED: true,
+    GOOGLE_STORAGE_PROJECT_ID: "blobscan",
+    GOOGLE_STORAGE_BUCKET_NAME: "blobscan-test",
+    GOOGLE_STORAGE_API_ENDPOINT: "http://localhost:8080",
+  },
+}));
+
+describe("Indexer route", async () => {
+  const caller = await getCaller();
+  const callerWithClient = await getCaller({ withClient: true });
+
+  describe("getSlot", () => {
+    it("should return the current slot", async () => {
+      const result = await caller.indexer.getSlot();
+
+      expect(result).toMatchObject({ slot: 101 });
+    });
+  });
+
+  describe("updateSlot", () => {
+    it("should not update the slot if not auth", async () => {
+      type Input = inferProcedureInput<AppRouter["indexer"]["updateSlot"]>;
+      const input: Input = {
+        slot: 110,
+      };
+
+      await expect(caller.indexer.updateSlot(input)).rejects.toThrow(
+        new TRPCError({ code: "UNAUTHORIZED" })
+      );
+    });
+
+    it("should update the slot if auth", async () => {
+      type Input = inferProcedureInput<AppRouter["indexer"]["updateSlot"]>;
+      const input: Input = {
+        slot: 110,
+      };
+
+      await callerWithClient.indexer.updateSlot(input);
+
+      const result = await caller.indexer.getSlot();
+
+      expect(result).toMatchObject({ slot: 110 });
+    });
+  });
+
+  describe("indexData", () => {
+    it("should index new data if auth", async () => {
```

Comment:
I think we should test the following cases as well:

1. Call the index procedure with repeated blobs as part of the input.
2. Check if the procedure is idempotent.